scss/grid/_layout.scss

```diff
@@ -468,8 +468,13 @@
         }
         .k-header-column-menu {
             margin: 0;
-            margin-top: calc(#{$grid-form-component-offset} * -1);
-            margin-bottom: calc(#{$grid-form-component-offset} * -1);
+            margin-top: (-$button-padding-y);
+            margin-bottom: (-$button-padding-y);
```

Comment:
Actually nested calc() functions with nested "calc()" string does not work in IE, so using this solution does not work in that browser. I set the "$button-padding-y" margin because of margin of  the ".k-link" next to the column menu, has negative margin with the same value as its padding:
- $cell-padding-y + ( $cell-padding-y - $button-padding-y ) = -$button-padding-y
where the first one is margin-top of ".k-link", and the expression in the brackets is difference of ".k-link" and ".k-header-column-menu" top padding.